subprojects/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -108,7 +129,30 @@ public void setType(String type) {
         return getProjectLayoutRegistry().getSupportedTypes();
     }
 
-    @Option(option = "test-framework", description = "Set alternative test framework to be used.", order = 1)
+    /**
+     * Set alternative build script DSL to be used.
+     *
+     * @since 4.5
+     */
+    @Incubating
+    @Option(option = "build-script-dsl", description = "Set alternative build script DSL to be used.", order = 1)
+    public void setBuildScriptDsl(String buildScriptsDsl) {
+        this.buildScriptDsl = buildScriptsDsl;
+    }
+
+    /**
+     * Available build script DSLs to be used.
+     *
+     * @since 4.5
+     */
+    @Incubating
+    @OptionValues("build-script-dsl")
+    @SuppressWarnings("unused")
+    public List<String> getAvailableBuildScriptDSLs() {
```

Comment:
Can't this be a `static` method? Why make it a method on the instance?

Reply:
By convention related to internal `@OptionValues`

Reply:
I don't understand. What do you mean?